server/route.go

```diff
@@ -632,18 +862,22 @@ func (c *client) sendRouteSubOrUnSubProtos(subs []*subscription, isSubProto bool
 		if filter != nil && !filter(sub) {
 			continue
 		}
-		rsid := routeSid(sub)
-		// Check if proto is going to fit.
-		curSize := len(buf)
-		if isSubProto {
-			// "SUB " + subject + " " + queue + " " + ...
-			curSize += 4 + len(sub.subject) + 1 + len(sub.queue) + 1
+		// Determine the account. If sub has an ImportMap entry, use that, otherwise scoped to
+		// client. Default to global if all else fails.
+		var accName string
+		if sub.im != nil {
+			accName = sub.im.acc.Name
+		} else if sub.client != nil && sub.client.acc != nil {
+			accName = sub.client.acc.Name
 		} else {
-			// "UNSUB " + ...
-			curSize += 6
+			accName = globalAccountName
 		}
-		// rsid + "\r\n"
-		curSize += len(rsid) + 2
+
+		// Check if proto is going to fit.
+		curSize := len(buf)
+		// "RS+/- " account + subject + " " + queue + " " + account
+		curSize += 4 + len(accName) + len(sub.subject) + 1 + len(sub.queue) + 2
```

Comment:
Should be:
```
// "RS+/- " + account + " " + subject + " " [+ queue + " " + weight] + CRLF
curSize += 4 + len(accName) + 1 + len(sub.subject) + 1 + len(sub.queue) + weightLen + 2
```
Queue is optional and we don't know how many char for weightLen, unless you move the computation of the weight string buffer before that. I don't think we need to be super accurate here, so you could overestimate.

Reply:
ok will fix.